packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5464,6 +5464,35 @@ describe('NetworkController', () => {
       );
     });
   });
+
+  describe('destroy', () => {
+    describe('if the blockTracker is defined', () => {
+      it('should stop the blockTracker', async () => {
+        await withController({}, async ({ controller }) => {
+          const fakeProvider = buildFakeProvider();
+          const fakeNetworkClient = buildFakeClient(fakeProvider);
+          createNetworkClientMock.mockReturnValue(fakeNetworkClient);
+          await controller.initializeProvider();
+          const destroySpy = jest.spyOn(
+            fakeNetworkClient.blockTracker,
+            'destroy',
+          );
+
+          await controller.destroy();
+
+          expect(destroySpy).toHaveBeenCalled();
+        });
+      });
+    });
+
+    describe('if the blockTracker is undefined', () => {
+      it('should not throw errors', async () => {
+        await withController({}, async ({ controller }) => {
+          expect(async () => await controller.destroy()).not.toThrow();
+        });
+      });
+    });
+  });
```

Comment:
I'm not sure of which test cases we want to cover for this method, and what exactly we should observe / assert.
For now, it seemed logical to check if the `blockTracker.destroy` function has been called when `blockTracker` is defined, and that no error is thrown when it isn't.

Reply:
These look similar to the tests we have in the extension for this method as well! Agreed that these is what we want to cover.